build-logic/lifecycle/src/main/kotlin/gradlebuild.lifecycle.gradle.kts

```diff
@@ -51,24 +48,11 @@ tasks.registerEarlyFeedbackRootLifecycleTasks()
  * Print all stacktraces of running JVMs on the machine upon timeout. Helps us diagnose deadlock issues.
  */
 fun setupTimeoutMonitorOnCI() {
-    if (BuildEnvironment.isCiServer && project.name != "gradle-kotlin-dsl-accessors") {
-        val timer = Timer(true).apply {
-            schedule(
-                timerTask {
-                    exec {
-                        commandLine(
-                            "${System.getProperty("java.home")}/bin/java",
-                            project.layout.projectDirectory.file("subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/timeout/JavaProcessStackTracesMonitor.java").asFile,
-                            project.layout.projectDirectory.asFile.absolutePath
-                        )
-                    }
-                },
-                determineTimeoutMillis()
-            )
-        }
-        gradle.buildFinished {
-            timer.cancel()
-        }
+    if (BuildEnvironment.isCiServer) {
+        project.gradle.sharedServices.registerIfAbsent("printStackTracesOnTimeoutBuildService", PrintStackTracesOnTimeoutBuildService::class.java) {
+            parameters.timeoutMillis.set(determineTimeoutMillis())
+            parameters.projectDirectory.set(layout.projectDirectory)
+        }.get().startTimer()
```

Comment:
Do you know if we start the timer as well when we detect the accessors? This happens before we compile the build script which applies the plugin and that is why we had `project.name != "gradle-kotlin-dsl-accessors"` before.

Reply:
I wasn't aware of that. I added it back in case.